src/IK/src/GravityTask.cpp

```diff
@@ -163,46 +147,35 @@ bool GravityTask::update()
 
     m_isValid = false;
 
-    // Normalize Accelerometer vector
-    m_accDenomNorm = sqrt(pow(m_currentAcc(0),2) + pow(m_currentAcc(1),2) + pow(m_currentAcc(2),2));
-
-    for (size_t i = 0; i < m_currentAcc.size(); i++)
-    {
-        m_currentAccNorm(i) = m_currentAcc(i) / m_accDenomNorm;
-    }
+    auto targetRotation = toEigen(m_kinDyn->getWorldTransform(m_targetFrameIndex).getRotation());
 
-    m_currentAccNorm = toEigen(m_kinDyn->getWorldTransform(m_targetFrameName).getRotation()) * m_currentAccNorm;
+    Eigen::Vector3d desiredZDirectionAbsolute = targetRotation * m_desiredZDirectionBody;
+    Eigen::Vector3d feedforwardAbsolute = targetRotation * m_feedForwardBody;
 
-    // get the relative jacobian
-    if (!m_kinDyn->getRelativeJacobian(m_baseIndex, m_targetFrameIndex, m_relativeJacobian))
```

Comment:
Why you removed the base frame, `m_baseName`,? The relative jacobian in my case computes the jacobian between the `Right Palm` frame and the fingertip frame. If not specified, it computes the jacobian between the robot base, in our case the `Pelvis` and the fingertip frame.

Reply:
> Why you removed the base frame, `m_baseName`,? The relative jacobian in my case computes the jacobian between the `Right Palm` frame and the fingertip frame. If not specified, it computes the jacobian between the robot base, in our case the `Pelvis` and the fingertip frame.

I have given it thought again and apparently, also in my case, there is no need to compute the Jacobian between the fingertip and the `RightPalm`. Probably, I confused it with the DistanceTask where it was important to assume the `RightPalm` as the base.